pkg/agent/runner.go

```diff
@@ -1314,6 +1309,21 @@ func (r *Runner) doVMUpdate(
 		r.vm.SetUsing(amount)
 	}
 
+	invalidInformantErr := func() error {
+		r.lock.Lock()
+		defer r.lock.Unlock()
+
+		if r.server == nil {
+			return errors.New("no informant server set")
+		}
+		return r.server.Valid()
```

Comment:
nit: I'd make a method on `r` that did all of this rather than defining it in a function here

Reply:
Done